apollo-ios/Sources/Apollo/FieldPolicyEvaluator.swift

```diff
@@ -0,0 +1,190 @@
+import Foundation
+import ApolloAPI
+
+enum FieldPolicyResult {
+  case single(CacheKeyInfo)
+  case list([CacheKeyInfo])
+}
+
+struct FieldPolicyEvaluator {
+  let field: Selection.Field
+  let variables: GraphQLOperation.Variables?
+  
+  init(
+    field: Selection.Field,
+    variables: GraphQLOperation.Variables?
+  ) {
+    self.field = field
+    self.variables = variables
+  }
+  
+  func resolveFieldPolicy() -> FieldPolicyResult? {
+    guard let fieldPolicy = field.fieldPolicy,
+          let arguments = field.arguments else {
+      return nil
+    }
+    
+    struct ParsedKey { let name: String; let path: [String]? }
+
+    let parsed: [ParsedKey] = fieldPolicy.keys.map { key in
```

Comment:
While this looks fine here, as I'm reading the rest of the code, the mental overhead of translating `parsed` to "this is the list of the keyArgs of the fieldPolicy` is too much. It makes the code hard to read.

```suggestion
    let keyArgs: [ParsedKey] = fieldPolicy.keys.map { key in
```